### PR TITLE
Add type hints to interfaces

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Added
- Add type hints to interfaces
```

### CODE REVIEW

**Review by VincentLanglet — APPROVED** [review]

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -15,6 +15,11 @@ parameters:
 			count: 1
 			path: src/Document/BaseDocumentManager.php
 
+		-
+			message: "#^Method Sonata\\\\Doctrine\\\\Model\\\\BaseManager\\:\\:create\\(\\) should return T of object but returns mixed\\.$#"
```
Comment (VincentLanglet) [inline phpstan-baseline.neon:19]: Do you know why we have this new error ?

**Review by VincentLanglet — DISMISSED** [review]

File: phpstan-baseline.neon (hunk 1 above)
Comment (wbloszyk) [reply phpstan-baseline.neon:19]: Becouse `return new $this->class();`. 
phpStan do not know what class is `$this->class`. W must say what is is:
```php
/**
     * @var string
     * @phpstan-var class-string<T> // add this line
     */
    protected $class;
```

**Review by wbloszyk — CHANGES_REQUESTED** [review]
I add in previus comment what to do.

File: phpstan-baseline.neon (hunk 1 above)
Comment (wbloszyk) [reply phpstan-baseline.neon:19]: @core23 Do you have time ti fix it?

**Review by phansys — APPROVED** [review]

**Review by wbloszyk — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Comment by jordisala1991** [comment]
Thank you @core23

**Pull request merged**